tests/e2e/e2e_fluentbit_out_test.go

```diff
@@ -0,0 +1,89 @@
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	. "github.com/openservicemesh/osm/tests/framework"
+)
+
+var _ = OSMDescribe("Test deployment of Fluent Bit sidecar",
+	OSMDescribeInfo{
+		Tier:   2,
+		Bucket: 2,
+	},
+	func() {
+		Context("FluentbitOutput", func() {
+			It("Forwards correctly filtered logs to stdout", func() {
+				// Install OSM with Fluentbit
+				installOpts := Td.GetOSMInstallOpts()
+				installOpts.DeployFluentbit = true
+				installOpts.SetOverrides = []string{"OpenServiceMesh.fluentBit.logLevel=info"}
+				Expect(Td.InstallOSM(installOpts)).To(Succeed())
+
+				pods, err := Td.Client.CoreV1().Pods(Td.OsmNamespace).List(context.TODO(), metav1.ListOptions{
+					LabelSelector: labels.SelectorFromSet(map[string]string{"app": "osm-controller"}).String(),
+				})
+				Expect(err).NotTo(HaveOccurred())
+
+				// Query fluentbit-logger container logs to test if Fluent Bit filters are working
+				for _, pod := range pods.Items {
+					// Wait until osm-controller has generated a log to check against
+					wait.Poll(time.Second*3, time.Second*180, isLogEmitted(pod.Namespace, pod.Name, "osm-controller", "\"level\":\"info\""))
+					for _, container := range pod.Spec.Containers {
+						if strings.Contains(container.Image, "fluent-bit") {
```

Comment:
@nojnhuh a little concerned if this `if` condition is not met, the test will pass since all the relevant expects are nested here

Reply:
That's true. I suppose you could have some sort of `fluentBitContainerFound` variable outside the loop, set it to true when you find it, then fail after the loop if it's false.

Reply:
ooh yeah sounds good